### PR TITLE
Fix: Execute build steps on Travis in appropriate sections

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] runs `composer self-update` in `before_install` section
* [x] runs `composer install` in `install` section

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
:+1:

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +14,10 @@ matrix:
     - php: 7.0
   fast_finish: true
 
-before_script:
+before_install:
   - composer self-update
```
Comment (GrahamCampbell) [inline .travis.yml]: :-1:

Comment (GrahamCampbell) [inline .travis.yml]: This should just be removed.

Comment (localheinz) [inline .travis.yml]: @GrahamCampbell 

It's actually quite useful if you want to make use of latest features - which may not be available on Travis yet, see for example #158.

Alternatively, we could check in `composer.phar`.

**Pull request merged**